e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Upload Files', () => {
+    let testTeam;
+
+    before(() => {
+        // # Create new team and new user and visit Town Square channel
+        cy.apiInitSetup({loginAfter: true}).then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T336 Image thumbnail - expanded RHS', () => {
+        const filename = 'huge-image.jpg';
+        const originalWidth = 1920;
+        const originalHeight = 1280;
+        const aspectRatio = originalWidth / originalHeight;
+        cy.log(aspectRatio);
+
+        // # Post an image in center channel
+        cy.get('#centerChannelFooter').find('#fileUploadInput').attachFile(filename);
+        cy.contains(filename).should('be.visible');
+        cy.postMessage('{enter}');
+
+        // # Click reply arrow to open the reply thread in RHS
+        cy.clickPostCommentIcon();
+
+        cy.get('#rhsContainer').within(() => {
+            // # Observe image thumbnail displays the same
+            cy.get('img[src*="/preview"]').should('be.visible').and((img) => {
+                expect(img.width() / img.height()).to.be.closeTo(aspectRatio, 0.01);
+            });
+
+            // # In the RHS, click the expand arrows to expand the RHS
+            cy.findByLabelText('Expand').click();
+
+            // * Observe image thumnail displays the same
+            cy.wait(TIMEOUTS.THREE_SEC);
```

Comment:
We need to wait until the image is expanded before we fetch its width and height. Is there a better way to wait until the animation is finished?